_aicspylibczi/Reader.h

```diff
@@ -66,265 +66,262 @@
+   */
+  SubblockMetaVec readSubblockMeta(libCZI::CDimCoordinate& plane_coord_, int index_m_ = -1);
+
+  /*!
+   * @brief provide the subblock info logicalRect coordinates
+   * @param plane_coord_ A structure containing the Dimension constraints
+   * @param index_m_ Must be set to select the tile of the mosaic file
+   * @return an IntRect {x, y, w, h}
+   */
+  libCZI::IntRect readSubblockRect(libCZI::CDimCoordinate& plane_coord_, int index_m_ = -1);
+
+  /*!
+   * @brief If the czi file is a mosaic tiled image this function can be used to reconstruct it into an image.
+   * @param plane_coord_ A class constraining the data to an individual plane.
+   * @param scale_factor_ (optional) The native size for mosaic files can be huge the scale factor allows one to get
+   * something back of a more reasonable size. The default is 0.1 meaning 10% of the native image.
+   * @param im_box_ (optional) The {x0, y0, width, height} of a sub-region, the default is the whole image.
+   * @return an ImagesContainerBasePtr containing the raw memory, a list of images, and a list of corresponding
+   * dimensions
+   *
+   * @code
+   *    FILE *fp = std::fopen("mosaicfile.czi", "rb");
+   *    czi = pylibczi::Reader(fp);
+   *    auto dims = czi.read_dims();  // dims = {{DimensionIndex::T, {0, 10}}, {DimensionIndex::C, {0,3}}
+   *    // the dims above mean that T and C need to be constrained to single values
+   *
+   *    auto c_dims = CDimCoordinate{ { DimensionIndex::T,0 }, { DimensionIndex::C,1 } };
+   *    auto img_p = czi.read_mosaic( c_dims, scaleFactor=0.15 );
+   *    if(img_p->is_type_match<uint16_t>(){ // supported types are uint8_t, uint16_t, and float
+   *        std::shared_ptr<uint16_t> img = img_p->get_derived<uint16_t>(); // cast the image to the memory size it was
+   * acquired as
+   *        // do what you like with the Image<uint16_t>
+   *    }
+   * @endcode
+   */
+  ImagesContainerBase::ImagesContainerBasePtr readMosaic(libCZI::CDimCoordinate plane_coord_,
+                                                         float scale_factor_ = 1.0,
+                                                         libCZI::IntRect im_box_ = { 0, 0, -1, -1 });
+  // changed from {.w=-1, .h=-1} to above to support MSVC and GCC - lagging on C++14 std
+
+  /*!
+   * Convert the libCZI::DimensionIndex to a character
+   * @param di_, The libCZI::DimensionIndex to be converted
+   * @return The character representing the DimensionIndex
+   */
+  static char dimToChar(libCZI::DimensionIndex di_) { return libCZI::Utils::DimensionToChar(di_); }
+
+  bool shapeIsConsistent() const { return !m_specifyScene; }
+
+  virtual ~Reader() { m_czireader->Close(); }
+
+  /*!
+   * Get the full size of the mosaic image without scaling. If you're selecting a sub-region it must be within the box
+   * returned.
+   * @return an IntRect {x, y, w, h}
+   */
+  libCZI::IntRect mosaicShape() const { return m_statistics.boundingBoxLayer0Only; }
```

Comment:
random thing i am noticing: is this guaranteed accurate?  I sort of recall it as being a short cut (i mean using boundingBoxLayer0Only)

Reply:
I'm not sure what you mean here. `boundingBoxLayer0Only` just means don't give me the bounding box of Non-Zero pyramid levels ie give me the data that was acquired not a pyramid level that was computationally generated.
To the best of my knowledge, there is no reason to believe `boundingBoxLayer0Only` would be inaccurate

Reply:
I reread the docs and though they are thin I believe the bounding box will give the dims of the mosaic file. 
The previous issue was using boundBoxLayer0Only or other libCZI::IntRects in calculations resulted in a loss of precision and overflow in the calculation